### PR TITLE
feat: Add option to choose mode for display file coverage

### LINKED ISSUE #160: Split the coverage report
Is there any chance we could split the coverage report between the files changed and the whole test suite?

### PR DESCRIPTION
**Pull request by sfer23**
Fixes #160, #190

### CODE REVIEW

File: src/getPullChanges.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+  // Skip Changes collection if we can't do it
+  if (!['pull_request', 'pull_request_review', 'pull_request_review_comment'].includes(github.context.eventName)) {
+    return [];
+  }
+
+  const gitHubToken = core.getInput('github-token').trim();
+  const prNumber = github.context.payload.pull_request.number
+  try {
+    const client = new github.GitHub(gitHubToken)
```
Comment (davelosert) [inline src/getPullChanges.ts:19]: You'll have to create the octokit-client like this:

```suggestion
    const octokit: Octokit = github.getOctokit(gitHubToken);
```

File: src/getPullChanges.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        per_page
+      })
+    ) as AsyncIterableIterator<any>) {
+      if (response.status !== 200) {
+        throw new Error(`Fetching list of changed files from GitHub API failed with error code ${response.status}`)
+      }
+      core.info(`Received ${response.data.length} items`)
+
+      for (const row of response.data) {
+        core.info(`[${row.status}] ${row.filename}`)
```
Comment (davelosert) [inline src/getPullChanges.ts:38]: Could you change this to `core.debug`? Else this will clutter up a lot of logs :)

File: src/index.ts [hunk 3]
```diff
@@ -3,15 +3,28 @@ import path from 'node:path';
...
+    // Do stuff here
+  }
+  const summaryFilesModeRaw = core.getInput('summary-mode');//'none'; // mixed/changes/all/none
+  let summaryFilesMode = summaryFilesModeRaw
+  if (!Object.values(SummaryModes).includes(summaryFilesModeRaw)) {
+    core.warning(`Not valid value "${summaryFilesModeRaw}" for summary mode, used "mixed"`)
+    summaryFilesMode = SummaryModes.Mixed
+  }
+
+  const pullChanges = getPullChanges(summaryFilesMode);
```
Comment (davelosert) [inline src/index.ts:27]: Missing await here:

```suggestion
  const pullChanges = await getPullChanges(summaryFilesMode);
```

**Review by davelosert — CHANGES_REQUESTED** [review]
Hi @sfer23, first of all thank you very much for opening this PR 🙇🏻.

It looks really great already! 👍🏻  

There are a few typing errors and some areas that require a little bit of brushing up, so I left some comments to address them, and with suggestions where it makes sense.

### Option `mixed`

As written in one of them, I think an option of `mixed` is not really required. I like the `all` mode (first showing the changes, then all other files) either way, so I'd make it the default. I even thought about making `changes` the default behavior, but I am a bit inconclusive as it would be a breaking change.

### Missing: Make table conditional if mode is set to `none` or no changes

What's still missing in this PR, but I think would be important is to **not** generate / show the coverage table if the Mode is set to `none` or if it is set to `changes` - but there are none (e.g. if only untested areas of the code were changed or of it is not in the context of a PR). I guess the right place to address this would be in the `index.js`. Would you mind putting that in?

### Hand over?

Last but not least: I really, really appreciate the work you already put into this and don't want to take up more of the time if you don't want to. So if you want me to take over, just let me know and I'll happily do the remaining work. Of course, I'll squash it all into your commit in the end, so you'll get the contribution any ways 🙂. 

Of course, if you want to take it to the finish line, I 💯% will let you do it. 🙂

**Comment by sfer23** [comment]
Thanks for feedback, it's perfect, I will fix that, but I can take some time (estimate about next week)

**Comment by davelosert** [comment]
Hey @sfer23 , just quickly wanted to check in if you were able to work on this or need any support?
Don't want to pressure you here, just really want to see this feature land 🙂.

Again, if you want me to take over, no problem.

**Comment by davelosert** [comment]
@sfer23 : Just a friendly reminder. I know we all get busy and that's perfectly fine. As I want to see this feature land (just got another bug report in #214), I'd take over if you won't come to do it / react here until tomorrow if that's alright!

**Comment by sfer23** [comment]
@davelosert Thanks for reminder, it's really not enough hours in days :) Here is a few fixes, not sure about one moment with Octkokit, my IDE say it's will not work (because of plugins) so I keep it as it was. But anyway I think you have much more experience with Node and GitHub so feel free to fix that moment also.

I also removed 'mixed' mode, because I added it just for fully backward compatibility and if you also think that it's not needed - I removed it.

Here is nice repo where I take a look for some examples: [getsentry/paths-filter](https://github.com/getsentry/paths-filter)
Maybe it can helps.

It's really nice to have such kind of code review and I thankful for your time also to help me, Thanks.

**Comment by davelosert** [comment]
@sfer23: Great work, and I'll gladly finish up here! Just a minor thing: Could you grant me write access to your fork of the repository? Then I could finish the work on this remote branch of yours - else I'll have to open up a new PR (not a big deal, but I'd be nicer to finish it on this one 🙂 )

**Comment by davelosert** [comment]
@sfer23 : Thank you so much again for your work here. 🙂 
I did some adjustments to fix some bugs and cover some edge-cases and things seem to work now.

I will merge this in tomorrow and do some testing before releasing it, hopefully also tomorrow.
This PR will receive a comment from the semantic-release bot as soon as the release was done. 👍🏻

**Pull request merged**